Wrappers/Python/cil/framework/data_container.py

```diff
@@ -0,0 +1,1548 @@
+            Specify the axis or axes to calculate 'mean' along. Can be specified as
+            string(s) of dimension_labels or int(s) of indices
+            Default None calculates the function over the whole array
+        out : ndarray or DataContainer, optional
+            Provide an object in which to place the result. The object must have the correct dimensions and
+            (for DataContainers) the correct dimension_labels, but the type will be cast if necessary. See
+            `Output type determination <https://numpy.org/doc/stable/user/basics.ufuncs.html#ufuncs-output-type>`_ for more details.
+            Default is None
+
+        Returns
+        -------
+        scalar or DataContainer
+            The mean as a scalar or inside a DataContainer with reduced dimension_labels
+            Default is to accumulate and return data as float64 or complex128
+        """
+
+        if kwargs.get('dtype') is not None:
+            warnings.warn("dtype is ignored (auto-using float64 or complex128)", DeprecationWarning, stacklevel=2)
+
+        if numpy.isrealobj(self.array):
+            kwargs['dtype'] = numpy.float64
+        else:
+            kwargs['dtype'] = numpy.complex128
+
+        return self._directional_reduction_unary(numpy.mean, axis=axis, out=out, *args, **kwargs)
+
+    # Logic operators between DataContainers and floats
+    def __le__(self, other):
+        '''Returns boolean array of DataContainer less or equal than DataContainer/float'''
+        if isinstance(other, DataContainer):
+            return self.as_array()<=other.as_array()
+        return self.as_array()<=other
+
+    def __lt__(self, other):
+        '''Returns boolean array of DataContainer less than DataContainer/float'''
+        if isinstance(other, DataContainer):
+            return self.as_array()<other.as_array()
+        return self.as_array()<other
+
+    def __ge__(self, other):
+        '''Returns boolean array of DataContainer greater or equal than DataContainer/float'''
+        if isinstance(other, DataContainer):
+            return self.as_array()>=other.as_array()
+        return self.as_array()>=other
+
+    def __gt__(self, other):
+        '''Returns boolean array of DataContainer greater than DataContainer/float'''
+        if isinstance(other, DataContainer):
+            return self.as_array()>other.as_array()
+        return self.as_array()>other
+
+    def __eq__(self, other):
+        '''Returns boolean array of DataContainer equal to DataContainer/float'''
+        if isinstance(other, DataContainer):
+            return self.as_array()==other.as_array()
+        return self.as_array()==other
+
+    def __ne__(self, other):
+        '''Returns boolean array of DataContainer negative to DataContainer/float'''
+        if isinstance(other, DataContainer):
+            return self.as_array()!=other.as_array()
+        return self.as_array()!=other
+
+
+class ImageData(DataContainer):
```

Comment:
Can `ImageData` and `ImageGeometry` be pulled out from here?

Reply:
done (f712ea4)